py/profiling.c

```diff
@@ -0,0 +1,1046 @@
+
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+
+#if MICROPY_PY_SYS_TRACE
+
+#include "py/profiling.h"
+#include "py/objstr.h"
+#include "py/objfun.h"
+#include "py/emitglue.h"
+#include "py/runtime.h"
+#include "py/bc0.h"
+#include "py/gc.h"
+
+#if PROF_PRINT_INSTR
```

Comment:
This whole section looks like unused code.

Reply:
The PROF_PRINT_INSTR symbol is enabled on-demand in py/profiling.h . It is intended for debugging purposes only.

Reply:
Do you mean debugging the settrace feature?  It's a lot of code just for debugging, and also duplicates things from vm.c/showbc.c.  It would be good to at least have the addition of this debugging code as a separate commit, so it's easier to understand what's needed/not needed for the basic settrace implementation.